File: Advanced/ArchNavigationDemo/app/src/main/java/edu/cs4730/archnavigationdemo/Fragment_Three.java
package edu.cs4730.archnavigationdemo;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import edu.cs4730.archnavigationdemo.databinding.FragmentThreeBinding;

/**
 * This example receives arguments via the safe_arg version.  Note safe args are included in Project build.grade (not module).
 */
public class Fragment_Three extends Fragment {

    FragmentThreeBinding binding;

    public Fragment_Three() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentThreeBinding.inflate(inflater, container, false);
        binding.tvPassed.setText(Fragment_ThreeArgs.fromBundle(requireArguments()).getMessage());

        String stuff = "Data 2 is " + Fragment_ThreeArgs.fromBundle(requireArguments()).getNumber();
        binding.tvPassed2.setText(stuff);
        return binding.getRoot();
    }

}

File: Advanced/ArchNavigationDemo_kt/app/src/main/java/edu/cs4730/archnavigationdemo_kt/Fragment_Three.kt
package edu.cs4730.archnavigationdemo_kt

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.fragment.app.Fragment
import edu.cs4730.archnavigationdemo_kt.databinding.FragmentThreeBinding

/**
 * This example receives arguments via the safe_arg version.  Note safe args are included in Project build.grade (not module).
 */
class Fragment_Three : Fragment() {
    lateinit var binding: FragmentThreeBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentThreeBinding.inflate(inflater, container, false)

        binding.tvPassed.text = Fragment_ThreeArgs.fromBundle(requireArguments()).message

        val stuff = "Data 2 is " + Fragment_ThreeArgs.fromBundle(requireArguments()).number
        binding.tvPassed2.text = stuff
        return binding.root
    }
}